backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceComponentListener.java

```diff
@@ -0,0 +1,100 @@
+package backtraceio.library.breadcrumbs;
+
+import android.content.ComponentCallbacks2;
+import android.content.res.Configuration;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import backtraceio.library.enums.BacktraceBreadcrumbLevel;
+import backtraceio.library.enums.BacktraceBreadcrumbType;
+
+public class BacktraceComponentListener implements ComponentCallbacks2 {
+
+    private BacktraceBreadcrumbs backtraceBreadcrumbs;
+
+    public BacktraceComponentListener(BacktraceBreadcrumbs backtraceBreadcrumbs)
+    {
+        this.backtraceBreadcrumbs = backtraceBreadcrumbs;
+    }
+
+    private String getMemoryWarningString(final int level)
+    {
+        switch (level)
+        {
+            case ComponentCallbacks2.TRIM_MEMORY_UI_HIDDEN:
+                return "TRIM_MEMORY_UI_HIDDEN";
+            case ComponentCallbacks2.TRIM_MEMORY_RUNNING_MODERATE:
+                return "TRIM_MEMORY_RUNNING_MODERATE";
+            case ComponentCallbacks2.TRIM_MEMORY_RUNNING_LOW:
+                return "TRIM_MEMORY_RUNNING_LOW";
+            case ComponentCallbacks2.TRIM_MEMORY_RUNNING_CRITICAL:
+                return "TRIM_MEMORY_RUNNING_CRITICAL";
+            case ComponentCallbacks2.TRIM_MEMORY_BACKGROUND:
+                return "TRIM_MEMORY_BACKGROUND";
+            case ComponentCallbacks2.TRIM_MEMORY_MODERATE:
+                return "TRIM_MEMORY_MODERATE";
+            case ComponentCallbacks2.TRIM_MEMORY_COMPLETE:
+                return "TRIM_MEMORY_COMPLETE";
+            default:
+                return "Generic memory warning";
+        }
+    }
+
+    private BacktraceBreadcrumbLevel getMemoryWarningLevel(final int level)
+    {
+        switch (level)
+        {
+            case ComponentCallbacks2.TRIM_MEMORY_RUNNING_MODERATE:
+            case ComponentCallbacks2.TRIM_MEMORY_RUNNING_LOW:
+            case ComponentCallbacks2.TRIM_MEMORY_RUNNING_CRITICAL:
```

Comment:
Maybe we should treat `            case ComponentCallbacks2.TRIM_MEMORY_RUNNING_CRITICAL:` as a fatal instead? There is high chance that we will never get this event - and if we will be somehow available to get it... it should mean that our application will probably soon go down.

Reply:
Good point, will do!